structuralcodes/sections/_generic.py

```diff
@@ -478,6 +581,233 @@ def calculate_moment_curvature(
+        eps_p_lim = eps_n_b * (h - (y_n - y_p)) / h
+        eps_n = np.append(eps_n, np.zeros(n) + eps_n_b)
+        eps_p = np.append(eps_p, np.linspace(0, eps_p_lim, n, endpoint=False))
+        # Field 6: pivot on eps_n_y point or eps_n_b
+        n = 40  # 8
+        # If reinforced concrete section pivot on eps_n_y (default -0.002)
+        # otherwise pivot on eps_n_b (in top chord)
+        if self.section.geometry.reinforced_concrete:
+            z_pivot = y_n - (1 - eps_n_y / eps_n_b) * h
+            eps_p_6 = np.linspace(eps_p_lim, eps_n_y, n, endpoint=True)
+            eps_n_6 = (
+                -(eps_n_y - eps_p_6) * (z_pivot - y_n) / (z_pivot - y_p)
+                + eps_n_y
+            )
+        else:
+            eps_n_6 = np.zeros(n) + eps_n_b
+            eps_p_6 = np.linspace(eps_p_lim, eps_n_b, n, endpoint=False)
+        eps_n = np.append(eps_n, eps_n_6)
+        eps_p = np.append(eps_p, eps_p_6)
+
+        # rotate them
+        kappa_y = (eps_n - eps_p) / (y_n - y_p)
+        eps_a = eps_n - kappa_y * y_n
+        kappa_z = np.zeros_like(kappa_y)
+
+        # rotate back components to work in section CRS
+        T = np.array([[cos(theta), -sin(theta)], [sin(theta), cos(theta)]])
+        components = np.vstack((kappa_y, kappa_z))
+        rotated_components = T @ components
+        return np.column_stack((eps_a, rotated_components.T))
+
+    def calculate_nmm_interaction_domain(
+        self, n_theta: int = 32, n_axial: int = 20
+    ) -> s_res.NmmInteractionDomain:
+        """Calculates the NMM interaction domain."""
+        res = s_res.NmmInteractionDomain()
+        res.n_theta = n_theta
+        # TODO: if we want to resample and structure better data we can
+        # use n_axial as the number of discretizations in axial direction
+        res.n_axial = n_axial
+
+        # cycle for all n_thetas
+        thetas = np.linspace(0, np.pi * 2, n_theta)
+        # Initialize an empty array with the correct shape
+        strains = np.empty((0, 3))
+        for theta in thetas:
+            # Get ultimate strain profiles for theta angle
+            strain = self._compute_ultimate_strain_profiles(theta=theta)
+            strains = np.vstack((strains, strain))
+
+        # integrate all strain profiles
+        forces = np.zeros_like(strains)
+        for i in range(strains.shape[0]):
+            N, My, Mz, tri = (
+                self.integrator.integrate_strain_response_on_geometry(
+                    geo=self.section.geometry,
+                    strain=strains[i, :],
+                    tri=self.triangulated_data,
+                )
+            )
+            if self.triangulated_data is None:
+                self.triangulated_data = tri
+            forces[i, 0] = N
+            forces[i, 1] = My
+            forces[i, 2] = Mz
+
+        # Save to results
+        res.strains = strains
+        res.forces = forces
+
+        return res
+
+    def calculate_mm_interaction_domain(
```

Comment:
Same as for the result dataclass, could we say `num_theta` to avoid confusion with the axial force?